### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kdaily**
This is an attempt to get separate production and develop stacks for VEO-IBD Genie infrastructure. This parameterizes the VPCs and adds in CloudWatch logs and associated roles. It also adds in a new Batch Job template.

### CODE REVIEW

**Comment by kdaily** [comment]
I've gotten this as far as I can right now! Could I get some assistance, please?

File: config/develop/veoibd.yaml [hunk 1]
```diff
@@ -1,14 +1,10 @@
 ---
 template_path: veoibd.yaml
-stack_name: veoibd
+stack_name: veoibd-develop
```
Comment (zaro0508) [inline config/develop/veoibd.yaml]: You probably don't want to do, it's better to keep the same stack names in both AWS sandbox and AWS scicomp accounts for consistency. 

Also when  you change this it deploys another stack `veoibd-develop` to sandbox but also keeps the old one `veoibd` as well.  you should login to sandbox and manually remove the one you don't want.

File: config/prod/veoibd.yaml [hunk 2]
```diff
@@ -1,13 +1,10 @@
 ---
 template_path: veoibd.yaml
-stack_name: veoibd
+stack_name: veoibd-prod
```
Comment (zaro0508) [inline config/prod/veoibd.yaml]: same here, just keep it `veoibd`

File: templates/veoibd.yaml [hunk 3]
```diff
@@ -41,18 +68,6 @@ Resources:
             Action:
               - 'sts:AssumeRole'
       Path: "/"
```
Comment (zaro0508) [inline templates/veoibd.yaml:68]: you probably want to add `PublishToCloudwatchPolicy` to this role.

File: templates/veoibd.yaml [hunk 4]
```diff
@@ -83,30 +98,28 @@ Resources:
...
+              !Sub '${AWS::Region}-${VpcName}-PrivateSubnet2'
         MinvCpus: 0
         ImageId: !Ref ComputeImageId
         InstanceRole: !GetAtt BatchInstanceProfile.Arn
         InstanceTypes:
           - optimal
         DesiredvCpus: 8
         Tags:
-          "Name": !Sub '${AWS::StackName}-BatchComputeEnvironment'
+          "Name": !Sub '${AWS::Region}-${AWS::StackName}-BatchComputeEnvironment'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: this cloudformation export does not exist.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: templates/veoibd.yaml [hunk 5]
```diff
@@ -27,6 +31,29 @@ Parameters:
     ConstraintDescription: 'Must be an acceptable email address syntax(i.e. joe.smith@sagebase.org)'
 
 Resources:
+  MyCloudWatchLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub '${AWS::Region}-${AWS::StackName}-CloudWatchLogGroup'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: do not set name if not required.  let cloudformation manage names.

File: templates/veoibd.yaml [hunk 6]
```diff
@@ -27,6 +31,29 @@ Parameters:
...
+      RetentionInDays: 365
+
+  PublishToCloudwatchPolicy:
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyName: !Sub '${AWS::Region}-${AWS::StackName}-PublishToCloudwatchPolicy'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: same here with the name

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: templates/veoibd.yaml [hunk 7]
```diff
@@ -83,30 +98,28 @@ Resources:
...
           "OwnerEmail": !Ref OwnerEmail
   BatchJobQueue:
     Type: "AWS::Batch::JobQueue"
     Properties:
-      JobQueueName: !Join
-        - '-'
-        - - !Ref AWS::StackName
-          - "job"
-          - "queue"
+      JobQueueName: !Sub '${AWS::Region}-${AWS::StackName}-BatchJobQueue'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: you want this to be `!Sub '${AWS::Region}-${AWS::StackName}-BatchJobQueue'`

File: templates/veoibd.yaml (hunk 4 above)
Comment (zaro0508) [inline templates/veoibd.yaml]: you want this to be `!Sub ${AWS::StackName}-BatchComputeEnvironment`

File: templates/veoibd.yaml [hunk 8]
```diff
@@ -116,13 +129,9 @@ Resources:
     Type: 'AWS::Batch::JobDefinition'
     Properties:
       Type: container
-      JobDefinitionName: !Join
-        - '-'
-        - - !Ref AWS::StackName
-          - "job"
-          - "mainprocess"
+      JobDefinitionName: !Sub '${AWS::Region}-${AWS::StackName}-MainBatchJob'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: just remove region

File: templates/veoibd.yaml [hunk 9]
```diff
@@ -137,8 +146,33 @@ Resources:
           - ContainerPath: "/root/.synapseCache"
             ReadOnly: false
             SourceVolume: synapsecache
+  ValidateBatchJob:
+    Type: 'AWS::Batch::JobDefinition'
+    Properties:
+      Type: container
+      JobDefinitionName: !Sub '${AWS::Region}-${AWS::StackName}-ValidateBatchJob'
```
Comment (zaro0508) [inline templates/veoibd.yaml]: remove region

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: templates/veoibd.yaml (hunk 3 above)
Comment (kdaily) [reply templates/veoibd.yaml:68]: How?

File: templates/veoibd.yaml (hunk 5 above)
Comment (kdaily) [reply templates/veoibd.yaml]: The BatchJobQueue name is not required either. Should I remove that?

File: templates/veoibd.yaml (hunk 7 above)
Comment (kdaily) [reply templates/veoibd.yaml]: The BatchJobQueue name is not required either. Should I remove that?

**Comment by zaro0508** [comment]
yes, remove it

**Pull request closed without merging**